nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -6,27 +6,27 @@ let
   cfg = config.services.nextcloud;
   fpm = config.services.phpfpm.pools.nextcloud;
 
-  phpPackage = pkgs.php73;
-  phpPackages = pkgs.php73Packages;
+  phpPackage = pkgs.php74.buildEnv {
+    exts = pp: with pp; [
+      bcmath calendar curl exif ftp filter gd gettext gmp intl json ldap
+      mysqlnd opcache openssl pcntl pdo pdo_mysql pdo_odbc pdo_pgsql
+      pdo_sqlite pgsql readline session soap sodium sqlite3 zip zlib mbstring
+      posix ctype dom simplexml xmlreader xmlwriter
+      apcu redis memcached imagick
```

Comment:
I'd assume these are the default extensions, plus the additional `acpu`, `redis`, `memcached` and `imagick` extensions - or did you go through the requirement list of nextcloud?

If it's just the default extensions, plus some additional ones, could we expose the default ones, maybe via `pkgs.php74.defaultExtensions`? That'd make things way more obvious.

I know it's less explicit, but should

Reply:
You're exactly right. It's the default list plus apcu, redis, memcached and imagick.

After we get this in I could make that available and document how to extend PHP based on the default set. I think our changeset is big enough as it is :-)

Reply:
@flokli How would you like a `php.withAdditionalExtensions` or `php.withExtraExtensions` that takes the default list and adds extra ones? :)

Of course we could expose the default list as well.

Reply:
Disclaimer: This disucssion should not block the PR. It's good to go IMHO!

Exposing the list of default extensions is probably already by itself quite readable and enough. A `withAdditionalExtensions` doesn't add too much benefit IMHO, and wouldn't allow me to do things like: take all the default extensions, but remove module x, and replace it with a patched version of it…

Reply:
I've added an `enabledExtensions` attribute and documentation about its use. I think this should be passed as an additional argument to `withExtensions` and `buildEnv.extensions`, optimally, but that would need much more work and this should suffice for now.